### PR TITLE
Bc buttons. add mute sound

### PR DESCRIPTION
**Pull request by NoSloppy**
Plays mute.wav if it exists before muted ignition to avoid confusion.

### CODE REVIEW

File: props/saber_BC_buttons.h [hunk 1]
```diff
@@ -490,12 +490,31 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
...
     if (scroll_presets_timer_.timerCheck() && scroll_presets_) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
   }  // Loop()
 
+
+  float GetSoundLengthNow(Effect* effect) {
+    RefPtr<BufferedWavPlayer> tmp = GetWavPlayerPlaying(effect);
+    if (tmp) {
+      tmp->UpdateSaberBaseSoundInfo();
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:510]: This is only needed when a sound is started automatically started, and we want to update the SaberBaseSoundInfo after the fact.  When you call PlayCommon(), the sound info will already be updated.

File: props/saber_BC_buttons.h [hunk 2]
```diff
@@ -775,15 +794,14 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
...
-    case EVENTID(BUTTON_POWER, EVENT_FOURTH_HELD, MODE_OFF):
-      if (!mode_volume_) {
-        if (SetMute(true)) {
-          unmute_on_deactivation_ = true;
-          scroll_presets_ = false;
-          On();
-        }
+case EVENTID(BUTTON_POWER, EVENT_FOURTH_HELD, MODE_OFF):
+    if (!mode_volume_) {
+      if (SFX_mute) {
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:799]: This will do nothing if there is no mute sound, I don't think that's what you want.

Comment (NoSloppy) [reply props/saber_BC_buttons.h:799]: ah I didn't test with no mute sound.
ok I'll be back.

File: props/saber_BC_buttons.h (hunk 1 above)
Comment (NoSloppy) [reply props/saber_BC_buttons.h]: ah. yes that's easier, thanks. changes made.

File: props/saber_BC_buttons.h (hunk 2 above)
Comment (NoSloppy) [reply props/saber_BC_buttons.h:791]: ok fixed.

**Pull request merged**